src/main/java/com/ericsson/ei/rules/MatchIdRulesHandler.java

```diff
@@ -42,10 +49,14 @@
      * */
     public List<String> fetchObjectsById(RulesObject ruleObject, String id) {
         String matchIdString = ruleObject.getMatchIdRules();
-        String fetchQueryString = replaceIdInRules(matchIdString, id);
-        MongoQuery fetchQuery = new MongoStringQuery(fetchQueryString);
-        List<String> objects = objHandler.findObjectsByCondition(fetchQuery);
-        return objects;
+        List<String> objects = new ArrayList<>();
+        try {
+            objects = doFetchObjectsById(matchIdString, id);
+            return objects;
+        } catch (ReplacementMarkerException e) {
+            LOGGER.warn("", e);
```

Comment:
Sorry missed this. Please put a comment on why you do like this.

Reply:
Check if the added comment is enough.